### PR TITLE
IpType validation moved to value method

### PR DESCRIPTION
**Pull request by pustovit**

### CODE REVIEW

File: sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java [hunk 1]
```diff
@@ -782,6 +783,15 @@ public void testInsertCoreTypesAsArray() throws Exception {
     }
 
     @Test
+    public void testInsertBadIpAdress() throws Exception {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java]: can you create a unit test instead of an integration test?

**Comment by dobe** [comment]
CHANGES update missing

File: core/src/test/java/io/crate/types/IpTypeTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            new BytesRef("123.34.243.23"),
+            new BytesRef("211.121.112.111"),
+            new BytesRef("0.0.0.0"),
+            new BytesRef("255.255.255.255")
+        };
+        for (BytesRef ip : validIps) {
+            assertEquals(true, ipType.isValid(ip));
+        }
+        BytesRef[] invalidIps = {
+                new BytesRef("192.168.0.2555"),
```
Comment (dobe) [inline core/src/test/java/io/crate/types/IpTypeTest.java]: pls add a test with two dots and also with dots at start and end

File: core/src/main/java/io/crate/types/IpType.java [hunk 3]
```diff
@@ -37,17 +37,25 @@ public int id() {
...
-            strIp = IpFieldMapper.longToIp(longIp);
+            String strIp = IpFieldMapper.longToIp(longIp);
+            return new BytesRef(strIp);
         }
-        return new BytesRef(strIp);
+    }
+
+    private void validate(BytesRef ip) {
+        if(!isValid(ip))
+            throw new IllegalArgumentException("Failed to validate ip");
```
Comment (dobe) [inline core/src/main/java/io/crate/types/IpType.java]: add the value to the msg

**Comment by dobe** [comment]
ack

**Comment by mfussenegger** [comment]
commited to 0.46 with https://github.com/crate/crate/commit/ca1127ed1dd1262d213fcc0a79c8900006caf1de

**Pull request closed without merging**